tests/System.Text.Primitives.Tests/Encoding/RandomTests.cs

```diff
@@ -147,8 +147,8 @@ public void ToString(string expected, Utf8String s)
         [Theory]
         [MemberData("StringEqualsTestCases_EmptyStrings")]
         [MemberData("StringEqualsTestCases_SimpleStrings")]
-        public unsafe void StringEquals(bool expected, Utf8String s1, Utf8String s2)
-        {
+        public unsafe void StringEqualsWithThreeArgs(bool expected, Utf8String s1, Utf8String s2)
+        {   // TODO: investigate why the tests fail if we have two methods with the same name StringEquals
```

Comment:
@KrzysztofCwalina Something changed in either XUnit or the compiler / tool-chain such that two tests with the exact same name but different arguments cause an issue.

In this case, there are 2 tests names 'StringEquals'. The bottom one (logical code order) takes 1 argument and has 2 inline test cases. For some reason, those test cases attempt to run using the 3 argument variant of the function and thus throw an exception at invocation time.

Renaming the function solves the problem, though, it might be worth filing a bug in any case. This seems like an odd behavior considering it worked before we upgraded to VS2017.